.circleci/config.yml

```diff
@@ -872,57 +872,14 @@ workflows:
   version: 2
   test-and-build:
     jobs:
-      # Fmt, vet, lint control plane and helm code
```

Comment:
I removed all of these jobs because they were duplicates of what is now in build-and-test.yml

Reply:
Q1: we have some lint jobs defined in GHA already (e.g. https://github.com/hashicorp/consul-k8s/blob/main/.github/workflows/golangci-lint-control-plane.yml). Should we move them into one pipeline? 

Q2: The golangci linter also runs go fmt and so you'd get two failures in our pipeline: one from linter and one from the `go-fmt-and-vet*` jobs. Should we just have the linter? I think we can configure it to also run go vet (https://pkg.go.dev/cmd/vet). I'm not sure if it already runs by default. This question maybe is a bit out-of-scope so it could make sense to address it in a separate PR if we decide to do so.

Reply:
@kschoche I followed the old workflow as much as possible. I lint early and try to parallel the jobs as much as possible. We can have more dependencies if we want to make sure all things stop right away if we want...


See: https://github.com/hashicorp/consul-k8s/actions/runs/1964296498